src/util/log.cc

```diff
@@ -529,6 +567,7 @@ options::options(program_options::option_group* parent_group)
     , logger_ostream_type(*this, "logger-ostream-type", logger_ostream_type::stderr,
             "Send log output to: none|stdout|stderr")
     , log_to_syslog(*this, "log-to-syslog", false, "Send log output to syslog.")
+    , log_with_color(*this, "log-with-color", true, "Print colored tag prefix in log message written to ostream")
```

Comment:
It's `true` by default and if a seastar app output is redirected to a file the latter becomes full of colorization marks. Any scripting/parsing built around it is affected badly. Can the option be `false` by default, and only if output is a terminal it turns on?

Reply:
@xemul thanks for the insights. i changed the default value of this option to `isatty(STDOUT_FILENO)`. does this make more sense? i am not quite sure how to implement your suggestion in the right way:

> Can the option be `false` by default, and only if output is a terminal it turns on?

we could have something like:

```c++
 log_with_color(*this, "log-with-color", false, "Print colored tag prefix in log message written to ostream")
...
void
logger::set_with_color(bool enabled) noexcept {
    fmt::formatter<log_level>::colored = enabled && isatty(STDOUT_FILENO);
}
```

but the question is whether we shall encode this "cleverness" in the code and override user settings?

Reply:
I think what the code has is fine. In any case this is just for developers, in production the logs will go via syslog.